File: account_rental_hub-frontend/src/services/transaction.service.js
import axios from "axios";

const API_URL = "http://localhost:8080/api/transaction";


const searchTransactions = async (page, size, customerName, userId, startDate, endDate, status) => {
  try {
    const response = await axios.get(`${API_URL}/search`, {
      params: {
        page,
        size,
        userId,
        customerName,
        status,
        startDate,
        endDate
      }
    });
    return response.data;
  } catch (error) {
    console.error("Error while searching account service:", error);
    throw error;
  }
}


const TransactionService = {
  searchTransactions,
}

export default TransactionService;

File: account_rental_hub-backend/src/main/java/com/AccountRentalHub/services/TransactionService.java
package com.AccountRentalHub.services;

import com.AccountRentalHub.models.Transaction;
import org.springframework.data.domain.Page;
import org.springframework.data.domain.Pageable;

import java.util.Date;

public interface TransactionService {
    Page<Transaction> searchTransactions(Long userId, String customerName, String status, Date startDate, Date endDate, Pageable pageable);
}
